packages/pigeon/lib/java_generator.dart

```diff
@@ -221,6 +221,11 @@ void _writeHostApi(Indent indent, Api api, Root root) {
             if (method.arguments.isNotEmpty) {
               indent.writeln(
                   'ArrayList<Object> args = (ArrayList<Object>)message;');
+              indent.write('if (args == null) ');
```

Comment:
What's the purpose of this change?

In general I want to move Pigeon toward trusting (or as @gaaclarke has proposed previously, validating during a single initial setup handshake) that the two sides are using the same generator, and that we therefore don't need to second-guess every data type and value because we control the generation of the code at both ends. Doing runtime checks on everything is needlessly inefficient, and this moving toward doing more of them instead of fewer.

Reply:
Yea, it's probably just best to ignore the warning with an annotation here if we can guarantee holistically that the value isn't null.  If we did want to turn this to an exception there should be a runtime test to assert it.

Reply:
I changed to use assert to remove this warning.
Or is it safe to use `@SuppressWarnings("ConstantConditions")` in class?